qiskit_nature/problems/second_quantization/electronic/electronic_structure_problem.py

```diff
@@ -57,8 +53,8 @@ def __init__(
 
     @property
     def num_particles(self) -> Tuple[int, int]:
-        molecule_data_transformed = cast(QMolecule, self._molecule_data_transformed)
-        return molecule_data_transformed.num_alpha, molecule_data_transformed.num_beta
+        particle_number = self._driver_result_transformed.get_property("ParticleNumber")
```

Comment:
I don't understand what you are saying here:
> I guess ones like the integrals we have data for the properties and can build out operators from them in order to measure them and get a result.
Also here:
> This ParticleNumber one, I guess the data to some degree is also the answer
What answer?

Are you asking about the fact that the number of expected particles (e.g., same for spin) may differ from the evaluated one? Yes, this can happen. But this is precisely the point. What we could do in the `interpret` method is raise a warning when this case occurs. This will immediately warn the user about such an occurrence.

Or are you asking about the general concept behind which `Property` objects exist?

Reply:
Unlike say the electronic energy or dipole property, which we cannot tell you what the property value is until its measured via the data the property contains, in the particle number property it has the number of particles from the get go. I am wondering if it will be obvious the difference, which to me is the number of particles we know to be in the molecule, versus the value that we will later measure which may not be the same.

Reply:
I see. But this is kind of the design which we have settled on now. This is kind-of the differentiation point between properties deriving the `IntegralProperty` class and those that don't. The former don't have previously known expected values whereas the latter do.
I think this can be well documented and should then be clear. Do you agree?

Reply:
Having properties that the user could choose what was computed by the stack is, as you say, what we wanted to have for flexibility, future expansion etc. In this type of case the property holds the actual value, so that can be used up front when we need to set something like num particles etc, but it will allow us to measure it later in the final solution to check the value there too. Yes, I agree documentation in the API ref, including noting this in some tutorial around using properties, should hopefully make this distinction clear enough for users.

Reply:
Okay I will make sure to improve the docs :+1: I want to include a tutorial either in the third PR for this or at a later point in time. Depending on available time/resources...